.github/workflows/ci_workflows.yml

```diff
@@ -16,61 +16,50 @@ jobs:
       fail-fast: true
       matrix:
         include:
-
-          - name: Python 3.7 with minimal dependencies
+          - name: Python 3.8 with minimal dependencies
             os: ubuntu-latest
-            python: 3.7
-            toxenv: py37-test
+            python: 3.8
+            toxenv: py38-test
 
-          - name: Python 3.8 with all optional dependencies
+          - name: Python 3.9 with all optional dependencies
             os: ubuntu-latest
-            python: 3.8
-            toxenv: py38-test-alldeps
+            python: 3.9
+            toxenv: py39-test-alldeps
             toxargs: -v --develop
 
-          # Commenting out the test below due to outdated ubuntu OS 
-          # (16.04) which is no longer supported, 
-          # and Python 3.6 is no longer being supported. 
-          # TODO: Might be worth keeping this test for users who 
-          # are still on 3.6 and just update the os to ubuntu-18.04
-          #- name: Python 3.6 with oldest supported version of all dependencies
-          #  os: ubuntu-16.04
-          #  python: 3.6
-          #  toxenv: py36-test-oldestdeps   
-            
-          - name: Python 3.7 with numpy 1.21 and full coverage
+          - name: Python 3.10 with numpy 1.23 and full coverage
             os: ubuntu-latest
-            python: 3.7
-            toxenv: py37-test-alldeps-numpy121-cov
-            
-          - name: Python 3.7 with all optional dependencies (MacOS X)
+            python: "3.10"
+            toxenv: py310-test-alldeps-numpy123-cov
+
+          - name: Python 3.10 with all optional dependencies (MacOS X)
             os: macos-latest
-            python: 3.7
-            toxenv: py37-test-alldeps
+            python: "3.10"
```

Comment:
because yaml would evaluate this to the number `3.1` without the quotes. and that is not a python release we want to be testing. so we force it to be a string `3.10` which gets us the correct version. it's not necessary for the other versions of python.